File: okex-php-sdk-api-v5/src/BrokerApi.php
<?php


namespace okv5;


class BrokerApi extends Utils
{
    public function info(){
        $params = [
        ];

        return $this->request('/api/v5/broker/nd/info', $params, 'GET');
    }

    public function createSubaccount($subAcct,$label='',$acctLv){
        $params = [
            'subAcct' => $subAcct,
            'label' => $label,
            'acctLv' => $acctLv,
        ];

        return $this->request('/api/v5/broker/nd/create-subaccount', $params, 'POST');
    }

    public function deleteSubaccount($subAcct){
        $params = [
            'subAcct' => $subAcct,
        ];

        return $this->request('/api/v5/broker/nd/delete-subaccount', $params, 'POST');
    }

    public function subaccountInfo($subAcct='',$page='',$limit=''){
        $params = [
            'subAcct' => $subAcct,
            'page' => $page,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/broker/nd/subaccount-info', $params, 'GET');
    }

    public function setSubaccountLevel($subAcct,$acctLv){
        $params = [
            'subAcct' => $subAcct,
            '$acctLv' => $acctLv,
        ];

        return $this->request('/api/v5/broker/nd/set-subaccount-level', $params, 'GET');
    }

    public function setSubaccountFeeRate($subAcct='',$instType='',$chgType,$chgTaker='',$chgMaker='',$effDate=''){
        $params = [
            'subAcct' => $subAcct,
            'instType' => $instType,
            'chgType' => $chgType,
            'chgTaker' => $chgTaker,
            'chgMaker' => $chgMaker,
            'effDate' => $effDate,
        ];

        return $this->request('/api/v5/broker/nd/set-subaccount-fee-rate', $params, 'POST');
    }

    public function subaccountDepositAddress($subAcct,$ccy,$chain='',$addrType='',$to=''){
        $params = [
            'subAcct' => $subAcct,
            'ccy' => $ccy,
            'chain' => $chain,
            'addrType' => $addrType,
            'to' => $to,
        ];

        return $this->request('/api/v5/asset/broker/nd/subaccount-deposit-address', $params, 'POST');
    }

    public function modifySubaccountDepositAddress($subAcct,$ccy,$chain='',$addr,$to){
        $params = [
            'subAcct' => $subAcct,
            'ccy' => $ccy,
            'chain' => $chain,
            'addr' => $addr,
            'to' => $to,
        ];

        return $this->request('/api/v5/asset/broker/nd/modify-subaccount-deposit-address', $params, 'POST');
    }

    public function getSubaccountDepositAddress($subAcct,$ccy){
        $params = [
            'subAcct' => $subAcct,
            'ccy' => $ccy,
        ];

        return $this->request('/api/v5/asset/broker/nd/subaccount-deposit-address', $params, 'GET');
    }

    public function subaccountDepositHistory($subAcct='',$ccy='',$txId='',$state='',$after='',$before='',$limit=''){
        $params = [
            'subAcct' => $subAcct,
            'ccy' => $ccy,
            'txId' => $txId,
            'state' => $state,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/asset/broker/nd/subaccount-deposit-history', $params, 'GET');
    }

    public function rebateDaily($subAcct='',$begin='',$end='',$page='',$limit=''){
        $params = [
            'subAcct' => $subAcct,
            'begin' => $begin,
            'end' => $end,
            'page' => $page,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/broker/nd/rebate-daily', $params, 'GET');
    }

    public function rebatePerOrders($type='',$begin='',$end=''){
        $params = [
            'type' => $type,
            'begin' => $begin,
            'end' => $end,
        ];

        return $this->request('/api/v5/broker/nd/rebate-per-orders', $params, 'GET');
    }

    public function PostrebatePerOrders($begin,$end){
        $params = [
            'begin' => $begin,
            'end' => $end,
        ];

        return $this->request('/api/v5/broker/nd/rebate-per-orders', $params, 'POST');
    }

    public function apikey($subAcct,$label,$passphrase,$ip='',$perm=''){
        $params = [
            'subAcct' => $subAcct,
            'label' => $label,
            'passphrase' => $passphrase,
            'ip' => $ip,
            'perm' => $perm,
        ];

        return $this->request('/api/v5/broker/nd/subaccount/apikey', $params, 'POST');
    }

    public function getApikey($subAcct,$apiKey=''){
        $params = [
            'subAcct' => $subAcct,
            'apiKey' => $apiKey,
        ];

        return $this->request('/api/v5/broker/nd/subaccount/apikey', $params, 'GET');
    }

    public function modifyApikey($subAcct,$apiKey,$label,$perm,$ip=''){
        $params = [
            'subAcct' => $subAcct,
            'apiKey' => $apiKey,
            'label' => $label,
            'perm' => $perm,
            'ip' => $ip,
        ];

        return $this->request('/api/v5/broker/nd/subaccount/modify-apikey', $params, 'POST');
    }

    public function deleteApikey($subAcct,$apiKey){
        $params = [
            'subAcct' => $subAcct,
            'apiKey' => $apiKey,
        ];

        return $this->request('/api/v5/broker/nd/subaccount/delete-apikey', $params, 'POST');
    }

}

File: okex-python-sdk-api-v5/okex/Broker_api.py
from .client import Client
from .consts import *


class BrokerAPI(Client):
    def __init__(self, api_key, api_secret_key, passphrase, use_server_time=False, flag='1'):
        Client.__init__(self, api_key, api_secret_key, passphrase, use_server_time, flag)

    def broker_info(self):
        params = {}
        return self._request_with_params(GET, BROKER_INFO, params)

    def create_subaccount(self, pwd='', subAcct='', label='', acctLv=''):
        params = {"pwd": pwd, 'subAcct': subAcct, 'label': label, 'acctLv': acctLv}
        return self._request_with_params(POST, CREATE_SUBACCOUNT, params)

    def delete_subaccount(self, pwd='', subAcct=''):
        params = {'pwd': pwd, 'subAcct': subAcct}
        return self._request_with_params(POST, DELETE_SUBACCOUNT, params)

    def subaccount_info(self, subAcct='', page='', limit=''):
        params = {'subAcct': subAcct, 'page': page, 'limit': limit}
        return self._request_with_params(GET, SUBACCOUNT_INFO, params)

    def set_subaccount_level(self, subAcct='', acctLv=''):
        params = {'subAcct': subAcct, 'acctLv': acctLv}
        return self._request_with_params(POST, SET_SUBACCOUNT_LEVEL, params)

    def set_subaccount_fee_rate(self, subAcct='', instType='', chgType='', chgTaker='', chgMaker='', effDate=''):
        params = {'subAcct': subAcct, 'instType': instType, 'chgType': chgType, 'chgTaker': chgTaker, 'chgMaker':chgMaker, 'effDate':effDate}
        return self._request_with_params(POST, SET_SUBACCOUNT_FEE_REAT, params)

    def subaccount_deposit_address(self, subAcct='', ccy='', chain='', addrType='', to=''):
        params = {'subAcct': subAcct, 'ccy': ccy, 'chain': chain, 'addrType': addrType, 'to': to}
        return self._request_with_params(POST, SUBACCOUNT_DEPOSIT_ADDRESS, params)

    def subaccount_deposit_history(self, subAcct = '', ccy = '', txId = '', state = '', after = '', before = '', limit = ''):
        params = {'subAcct': subAcct, 'ccy': ccy, 'txId': txId, 'state': state, 'after': after, 'before': before, 'limit':limit}
        return self._request_with_params(GET, SUBACCOUNT_DEPOSIT_HISTORY, params)

    def rebate_daily(self, subAcct = '', begin = '', end = '', page = '', limit = ''):
        params = {'subAcct': subAcct, 'begin': begin, 'end': end, 'page': page, 'limit': limit}
        return self._request_with_params(GET, REBATE_DAILY, params)